### ISSUE TITLE
Request: Add discovery data to nightly dumps for systems

### ISSUE THREAD

**Issue by NecroBones** [EDSM-NET/FrontEnd#195]
Hi there! I have a feature request. Right now, the nightly dumps of bodies has a "discovery" data set for each body, with the commander's name and discovery date. However, the systems don't have this, and the only date provided can change when the system is visited again later (presumably when the coordinates are first verified by a second visit). Would it be possible to add discovery data to the nightly dumps for systems? Thanks! :)

EDIT: I don't think this is included in the API lookups for systems either, which could also be useful, but I don't rely on those as much.

**Comment by klightspeed** [comment]
This was actually removed from the API and dumps in 2016, before any of the commander stuff was added (it used to be in the `submitted` json property of the system in the API).

**Comment by NecroBones** [comment]
Oh, was there a specific reason it was removed?

**Comment by NecroBones** [comment]
It occurs to me that if there's a privacy issue, leaving out the names is certainly fine. The main thing I really would like to add is the actual discovery/submission dates, since the date that's included now can change.

**Comment by AnthorNet** [comment]
We do not store submission date.
I'll check for the rest

**Comment by NecroBones** [comment]
Oh, I assumed that would be there since the website UI says "That system was first discovered for EDSM by <name> on <date>". The date isn't stored?

**Comment by AnthorNet** [comment]
The first discovery part is link on the scan event provided by player.
If another player is recording a previous scan, then that value is changed to the right owner.

**Comment by klightspeed** [comment]
I'm guessing the system's first discovery is based on the earliest travel log entry recorded for that system, and would be expensive to query for 20 million systems (basically a `MIN(log.date)` grouped by system ID)?

**Comment by AnthorNet** [comment]
You're right, I was on bodies ^^ But yes it works the same, by checking early logs. So when commanders add old logs, the discovery date is updated.

**Comment by NecroBones** [comment]
OK, that's not too bad. I don't mind the dates rolling backward for earlier discoveries. I'm just concerned about dates being updated forward because someone corroborated coordinates, or something, on a later visit, which the current date field in the dumps seems to do. It appears to behave as an "updated" date, which is great for detecting changes, but doesn't give as useful of a timeline information.

**Comment by NecroBones** [comment]
If it's requiring a huge join, then I understand why it's not there. :)

**Comment by AnthorNet** [comment]
It should be a simple JOIN, but of course it's not available on the main table.
I'll run some tests, but not sure when.

If you get our ID, those are incremental and do not change. If that helps.

**Comment by NecroBones** [comment]
OK great. It's not a high priority, I just noticed that since the dates sometimes change, it could have subtle impacts on the exploration history videos I've been generating. But I would think it would be rarely noticeable in most cases. I'm thinking ahead to any future correlations or analyses that I might do that I haven't thought up yet. :)
